### ISSUE TITLE
[BUG] Params.json not saved when using sf.model.Trainer

### ISSUE THREAD

**Issue by jamesdolezal** [jamesdolezal/slideflow#133]
## Description
`params.json` is not saved when using a `sf.model.Trainer` directly, which creates issues with attempting to evaluate a saved model.

## Environment:

 - Slideflow Version (e.g., 1.0): 1.0.4

**Comment by jamesdolezal** [comment]
128f658 moved the params.json saving from `Project` to `model.Trainer`. The expected behavior is now:

**Training with Project class**
_Typical, high-level training strategy_
- If training with `Project.train()` (typical behavior), no changes are made. params.json is saved automatically into project folder. Evaluation of saved models can be done with `Project.evaluate()` or manually with the `model.Trainer`.

**Training with model.Trainer**
_Low-level manual training strategy [advanced]_
- If training manually with `model.Trainer.train()`, evaluation can also be done manually with `model.Trainer.evaluate()`. Labels are manually provided with the argument `labels`. **The user is responsible for ensuring that label assignments match between the model training and model evaluation. While `Project.train()` does this automatically via the saved `params.json` file, this does not occur with the lower-level `model.Trainer` class.**
- If a dictionary is provided via the argument `Trainer(config=...)` (specifying the training configuration, outcome annotation headers, and outcome label assignment), this will be saved as params.json. 
- If `config` argument is not provided, the saved params.json file will default to only including the slideflow version and hyperparameters, and will not include any information about the outcome labels or label assignments. For this reason, the saved trained model **cannot be evaluated with Project.evaluate()**.
- This can be circumvented by manually providing `hp`, `outcome_label_headers`, and `outcome_labels` as a dictionary to the `config` argument during training, which provides enough information for the `Project.evaluate()` function to determine outcome label information from the project annotations file.

**Note:** we recognize that the above behavior for the `model.Trainer` is unintuitive. However, we discourage use of the `model.Trainer` class directly, as it circumvents several helpful features of `Project.train()` (such as automatic outcome label assigning and matching between training and evaluation). We have kept `model.Trainer` available as a lower-level interface for advanced projects requiring granular control.
